### PR TITLE
DecisionTreeRegressor added to pymilo param

### PR DESCRIPTION
**Pull request by AHReccese**
#### Reference Issues/PRs
https://github.com/openscilab/pymilo/issues/18

#### What does this implement/fix? Explain your changes.
Add full support for Decision Tree Regressor 

#### Any other comments?

### CODE REVIEW

**Review by sepandhaghighi — CHANGES_REQUESTED** [review]
@AHReccese Thanks for your efforts 💯 

1. Would you please update the `CHANGELOG.md`?
2. I am not sure about the docstring style, particularly when it comes to multi-line mode.
@alirezazolanvari would you please check it out?

**Comment by AHReccese** [comment]
> @AHReccese Thanks for your efforts 💯
> 
> 1. Would you please update the `CHANGELOG.md`?
> 2. I am not sure about the docstring style, particularly when it comes to multi-line mode.
>    @alirezazolanvari would you please check it out?

1. Changelog updated.
2. Could you please explain it a little bit more? what do you mean by docstring style? nothing is changed from the previous versions, I follow the same structure, if there is anything I can do let me know.

File: pymilo/chains/decision_tree_chain.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        setattr(raw_model, item, data[item])
+    return raw_model
+
+
+def _validate_input(model, command):
+    """
+    Check if the provided inputs are valid in relation to each other.
+
+    :param model: given object to gets transported, whether a sklearn decision tree model to get serialized
+    or a json string of a decision tree model to get deserialized to associated sklearn decision tree model
```
Comment (alirezazolanvari) [inline pymilo/chains/decision_tree_chain.py:112]: @AHReccese Please summarize the param description by including these details in the description line.

File: pymilo/transporters/tree_transporter.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+import platform
+
+class TreeTransporter(AbstractTransporter):
+    """Customized PyMilo Transporter developed to handle (pyi,pyx) Tree object."""
+
+    def serialize(self, data, key, model_type):
+        """
+        Serialize instances of the Tree class.
+
+        Record the n_features, n_classes and n_outputs fields of tree object.
```
Comment (alirezazolanvari) [inline pymilo/transporters/tree_transporter.py:19]: Remove the blank line between the summary descriptions. There should be one blank line between the summary and params sections.

File: pymilo/transporters/tree_transporter.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        """
+        Deserialize the special tree_ field of the Decision Trees.
+
+        The associated tree_ field of the pymilo serialized model, is extracted through
+        it's previously serialized parameters.
+
+        deserialize the data[key] of the given model which type is model_type.
+        basically in order to fully deserialize a model, we should traverse over all the keys of its serialized data dictionary and
+        pass it through the chain of associated transporters to get fully deserialized.
+
```
Comment (alirezazolanvari) [inline pymilo/transporters/tree_transporter.py:64]: Please remove the blank lines between the summary descriptions.

File: pymilo/transporters/tree_transporter.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        :param data: the internal data dictionary of the associated json file of the ML model which is generated previously by
```
Comment (alirezazolanvari) [inline pymilo/transporters/tree_transporter.py:65]: Please summarize it in a single line. My suggestion is: 
"the internal data dictionary of the associated JSON file of the ML model generated by pymilo export"

**Comment by alirezazolanvari** [comment]
> > @AHReccese Thanks for your efforts 💯
> > 
> > 1. Would you please update the `CHANGELOG.md`?
> > 2. I am not sure about the docstring style, particularly when it comes to multi-line mode.
> >    @alirezazolanvari would you please check it out?
> 
> 1. Changelog updated.
> 2. Could you please explain it a little bit more? what do you mean by docstring style? nothing is changed from the previous versions, I follow the same structure, if there is anything I can do let me know.

I left some minor comments. Please check them out and resolve them if they look fine with you.

File: pymilo/transporters/tree_transporter.py (hunk 2 above)
Comment (AHReccese) [reply pymilo/transporters/tree_transporter.py:18]: image

Comment (AHReccese) [reply pymilo/transporters/tree_transporter.py:20]: the first line is the `Summary` line and the second line is the description line and there should be one blank line between them. we can't have multi-line summary.

**Comment by AHReccese** [comment]
> > > @AHReccese Thanks for your efforts 💯
> > > 
> > > 1. Would you please update the `CHANGELOG.md`?
> > > 2. I am not sure about the docstring style, particularly when it comes to multi-line mode.
> > >    @alirezazolanvari would you please check it out?
> > 
> > 
> > 
> > 1. Changelog updated.
> > 2. Could you please explain it a little bit more? what do you mean by docstring style? nothing is changed from the previous versions, I follow the same structure, if there is anything I can do let me know.
> 
> I left some minor comments. Please check them out and resolve them if they look fine with you.

I applied all the feasible ones.

**Review by alirezazolanvari — APPROVED** [review]

File: pymilo/transporters/tree_transporter.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+"""PyMilo SGDOptimizer object transporter."""
```
Comment (sepandhaghighi) [inline pymilo/transporters/tree_transporter.py:2]: SGDOptimizer?!

File: CHANGELOG.md [hunk 6]
```diff
@@ -5,8 +5,18 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Added
+- scikit-learn decision tree models
+- `DecisionTreeRegressor` model
+- `Tree` Transporter
+- Decision Tree chain
+- `decision_tree_chain.py` 
+- `DecisionTreeRegressor` Test
```
Comment (sepandhaghighi) [inline CHANGELOG.md:14]: Please remove this line

File: CHANGELOG.md [hunk 7]
```diff
@@ -5,8 +5,18 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
...
 ### Changed
 - Tests config modified
+- DecisionTree params initialized in `pymilo_param`
+- Decision Tree support added `pymilo_func.py`
```
Comment (sepandhaghighi) [inline CHANGELOG.md:18]: added to

**Review by sepandhaghighi — CHANGES_REQUESTED** [review]
@AHReccese Take a look at my comments 🔥

File: CHANGELOG.md (hunk 6 above)
Comment (AHReccese) [reply CHANGELOG.md:14]: Do you mean that I wrote the Decision tree chain twice? hmm, you're right, I did remove the second one(`decision_tree_chain.py`)

**Review by sepandhaghighi — APPROVED** [review]
🚀

**Pull request merged**